core/src/main/java/org/apache/iceberg/DataFiles.java

```diff
@@ -352,7 +320,8 @@ public DataFile build() {
       return new GenericDataFile(
           filePath, format, isPartitioned ? partitionData.copy() : null,
           fileSizeInBytes, new Metrics(
-              recordCount, columnSizes, valueCounts, nullValueCounts, lowerBounds, upperBounds), keyMetadata);
+              recordCount, columnSizes, valueCounts, nullValueCounts, lowerBounds, upperBounds),
+                  keyMetadata, offsetRanges);
```

Comment:
Indentation should match the start of `recordCount`.

I think this should also copy the offset ranges using `ImmutableList`. There's no guarantee that it won't be reused or modified.